packages/vm/lib/runBlock.ts

```diff
@@ -149,17 +149,26 @@ export default async function runBlock(this: VM, opts: RunBlockOpts): Promise<Ru
   }
 
   // Checkpoint state
-  await state.checkpoint()
+  // (only do checkpointing when txs are executed)
+  // TODO: determine if checkpointing is needed in a block
+  // context at all or CPs in runTx() are sufficient
+  if (block.transactions.length > 0) {
+    await state.checkpoint()
+  }
   let result
   try {
     result = await applyBlock.bind(this)(block, opts)
   } catch (err) {
-    await state.revert()
+    if (block.transactions.length > 0) {
+      await state.revert()
+    }
```

Comment:
I guess this will work as I cannot think of a case where the block has 0 txs and throws (in that case, we'd need to rollback the `coinbase` account to the previous block)

Reply:
You know what? This extra case with the zero txs will get somewhat unnecessary with the better checkpointing mechanism anyhow and no need to take in extra risks here (haven't thought about the `coinbase` account e.g.), I will just remove this part of the commit.